protocol/v2/ssv/validator/committee.go

```diff
@@ -90,120 +89,89 @@ func (c *Committee) RemoveShare(validatorIndex phase0.ValidatorIndex) {
 	}
 }
 
-func (c *Committee) StartConsumeQueue(logger *zap.Logger, duty *spectypes.CommitteeDuty) error {
-	c.mtx.Lock()
-	defer c.mtx.Unlock()
-
-	// Setting the cancel function separately due the queue could be created in HandleMessage
-	q, found := c.Queues[duty.Slot]
-	if !found {
-		return errors.New(fmt.Sprintf("no queue found for slot %d", duty.Slot))
-	}
-
-	r := c.Runners[duty.Slot]
-	if r == nil {
-		return errors.New(fmt.Sprintf("no runner found for slot %d", duty.Slot))
-	}
-
-	// required to stop the queue consumer when timeout message is received by handler
-	queueCtx, cancelF := context.WithDeadline(c.ctx, time.Unix(c.BeaconNetwork.EstimatedTimeAtSlot(duty.Slot+runnerExpirySlots), 0))
-
-	go func() {
-		defer cancelF()
-		if err := c.ConsumeQueue(queueCtx, q, logger, duty.Slot, c.ProcessMessage, r); err != nil {
-			logger.Error("❗failed consuming committee queue", zap.Error(err))
-		}
-	}()
-	return nil
-}
-
 // StartDuty starts a new duty for the given slot
 func (c *Committee) StartDuty(logger *zap.Logger, duty *spectypes.CommitteeDuty) error {
-	c.mtx.Lock()
-	defer c.mtx.Unlock()
+	prepareRunner := func() (runner.Runner, error) {
```

Comment:
Since we touch it, can we extract it into another method for readability? Perhaps we could split this as well

Reply:
Yep, addressed in https://github.com/ssvlabs/ssv/pull/1784/commits/483c2ed41d9b789251d57e9d6ce4f3275e3f02c1